crates/broker/src/order_picker.rs

```diff
@@ -332,35 +359,47 @@ where
         }
 
         let one_mill = U256::from(1_000_000);
+        let total_cycles = U256::from(proof_res.stats.total_cycles);
+
+        let (min_price, max_price) = if slashed_unexpired {
+            let price = order.request.offer.lockStake / U256::from(FRACTION_STAKE_REWARD);
+            (price, price)
+        } else {
+            (U256::from(order.request.offer.minPrice), U256::from(order.request.offer.maxPrice))
+        };
```

Comment:
Since the stake token and regular price are different currencies, I think its best if we branch here into two separate paths of code to avoid confusion. There is also an issue in that currently provers specify "min_mcycle_price" in terms of ETH, so we can't use it to decide whether to pick an order where the price is in stake token. I think we'll need to introduce a prover config like "min_mcycle_price_stake" to handle this.

Reply:
Ok I switch the pricing strategy depending on the order type and added a new config field `mcycle_price_stake_token`. This is a fill-in until the broker can read the price of the stake token from somewhere